solutions/silver/usaco-895.mdx

```diff
@@ -19,74 +24,86 @@ import java.io.*;
 import java.util.*;
 
 public class IcyPerimeter {
-	public static void main(String[] args) throws IOException {
-		Kattio io = new Kattio("perimeter");
-		n = io.nextInt();
-
-		// we do n + 2 because there is an extra outer ring for flood fill
-		numberedMatrix = new int[n + 2][n + 2];
-
-		for (int x = 1; x <= n; x++) {  // read matrix
-			String line = io.next();
-			for (int y = 1; y <= n; y++) {
-				String character = Character.toString(line.charAt(y - 1));
-				if (character.equals("#")) {
-					numberedMatrix[x][y] = -1;  // mark # as -1
+	static int n;
+	static char[][] grid;
+	static boolean[][] visited;
+	static final int[][] directions = {
+	    {-1, 0},
+	    {1, 0},
+	    {0, -1},
+	    {0, 1}};  // Movement directions (up, down, left, right)
+
+	// BFS function to calculate the area and perimeter of a blob of ice cream
+	static int[] bfs(int x, int y) {
+		Queue<int[]> queue = new LinkedList<>();
+		queue.add(new int[] {x, y});  // Start BFS from the given cell (startX, startY)
+
+		int area = 0;
+		int perimeter = 0;
+
+		while (!queue.isEmpty()) {
+			int[] current = queue.poll();
+			int curX = current[0], curY = current[1];  // Current cell
```

Comment:
how about just `row` and `col`, and for `newX` and `newY` just `nextRow` and `nextCol` or something like that